api/tests/unit_tests/core/helper/test_url_signer.py

```diff
@@ -0,0 +1,179 @@
+from unittest.mock import patch
+
+import pytest
+
+from core.helper.url_signer import SignedUrlParams, UrlSigner
+
+
+class TestUrlSigner:
+    """Test cases for UrlSigner class"""
+
+    @patch("configs.dify_config.SECRET_KEY", "test-secret-key-12345")
+    def test_should_generate_signed_url_params(self):
+        """Test generation of signed URL parameters with all required fields"""
+        sign_key = "test-sign-key"
+        prefix = "test-prefix"
+
+        params = UrlSigner.get_signed_url_params(sign_key, prefix)
+
+        # Verify the returned object and required fields
+        assert isinstance(params, SignedUrlParams)
+        assert params.sign_key == sign_key
+        assert params.timestamp is not None
+        assert params.nonce is not None
+        assert params.sign is not None
+
+        # Verify nonce format (32 character hex string)
+        assert len(params.nonce) == 32
+        assert all(c in "0123456789abcdef" for c in params.nonce)
+
+    @patch("configs.dify_config.SECRET_KEY", "test-secret-key-12345")
+    def test_should_generate_complete_signed_url(self):
+        """Test generation of complete signed URL with query parameters"""
+        base_url = "https://example.com/api/test"
+        sign_key = "test-sign-key"
+        prefix = "test-prefix"
+
+        signed_url = UrlSigner.get_signed_url(base_url, sign_key, prefix)
+
+        # Verify URL contains all required parameters
+        assert signed_url.startswith(base_url)
```

Comment:
IMO it's better to parse the signed url and assert the query string contains `timestamp`, `nonce` and `sign`.